slack_sdk/oauth/installation_store/amazon_s3/__init__.py

```diff
@@ -60,6 +60,22 @@ def save(self, installation: Installation):
             )
             self.logger.debug(f"S3 put_object response: {response}")
 
+            # per workspace
+            entity: str = json.dumps(installation.__dict__)
+            response = self.s3_client.put_object(
+                Bucket=self.bucket_name,
+                Body=entity,
+                Key=f"{workspace_path}/installer-latest",
+            )
+            self.logger.debug(f"S3 put_object response: {response}")
+            response = self.s3_client.put_object(
+                Bucket=self.bucket_name,
+                Body=entity,
+                Key=f"{workspace_path}/installer-{history_version}",
+            )
+            self.logger.debug(f"S3 put_object response: {response}")
+
+            # per workspace per user
```

Comment:
The current `installation` data in files / Amazon s3 is per workspace per user. In this PR, I've added a new write operation to write without user_id in index key. The per-user files are still useful for fetching user's latest tinstallation (or all installations) with minimum cost.